CMakeLists.txt

```diff
@@ -70,6 +70,24 @@ if(NOT IS_SUBPROJECT)
   message(STATUS "-------- BUILD-TYPE: ${CMAKE_BUILD_TYPE} --------")
 endif()
 
+# Python Support
+option(PythonSupport "Build with Python support" ON)
+if(PythonSupport AND NOT TRIQS_WITH_PYTHON_SUPPORT)
+  message(FATAL_ERROR "TRIQS was installed without Python support. Cannot build the Python Interface.")
+endif()
+
+# Documentation
+option(Build_Documentation "Build documentation" OFF)
+if(Build_Documentation)
+    if(NOT TRIQS_WITH_DOCUMENTATION)
+      message(WARNING "TRIQS was installed without documentation! Cannot build documentation.")
+    endif()
+    if(IS_SUBPROJECT)
+      message(WARNING "Documentation building not supported for subproject.")
+    endif()
```

Comment:
I would drop these lines. Otherwise we will trigger one warning per SUBPROJECT / dependency that we build.